epoch-verifier/src/data/context.ts

```diff
@@ -1,36 +1,43 @@
-import { Address, Allocation, EpochInfo, Reward, UserBudget, UserDonation } from "./models"
+import {
+  Address,
+  Allocation,
+  AllocationRecord,
+  EpochInfo,
+  Reward,
+  UserBudget,
+  UserDonation
+} from "./models"
 
 export interface Context {
   allocations: Allocation[]
-  budgets: Map<Address, bigint> 
+  budgets: Map<Address, bigint>
   epochInfo: EpochInfo
   rewards: Reward[]
 }
 
-export function buildContext(budgets: UserBudget[], allocations: Allocation[], rewards: Reward[], epochInfo: EpochInfo): Context{
+export function buildContext(budgets: UserBudget[], allocations: AllocationRecord[], rewards: Reward[], epochInfo: EpochInfo): Context {
 
-  const filteredBudgets = budgets.filter((ub) => ub.budget !== BigInt(0))
-  const filteredAllocs = allocations.filter(alloc => alloc.donations.length !== 0)
+  const filteredBudgets = budgets.filter((ub) => ub.amount !== BigInt(0))
 
   const context = {
-    allocations: filteredAllocs,
-    budgets: new Map(filteredBudgets.map((value) => [value.user, value.budget] as const)),
+    allocations: transformToAllocations(allocations),
+    budgets: new Map(filteredBudgets.map((value) => [value.user, value.amount] as const)),
```

Comment:
Array of objects wouldn't be more handy instead of array of arrays? Array of arrays is faster to operate with, sure, but less readable.